oeommtools/utils.py

```diff
@@ -66,28 +66,45 @@ def oemol_to_openmmTop(mol):
                     oe_atom_to_openmm_at[oe_at] = openmm_at
 
     if topology.getNumAtoms() != mol.NumAtoms():
-        oechem.OEThrow.Fatal("OpenMM topology and OEMol number of atoms mismatching: "
-                             "OpenMM = {} vs OEMol  = {}".format(topology.GetNumAtoms(), mol.NumAtoms()))
+        oechem.OEThrow.Error("OpenMM topology and OEMol number of atoms mismatching: "
+                             "OpenMM = {} vs OEMol  = {}".format(topology.getNumAtoms(), mol.NumAtoms()))
 
     # Count the number of bonds in the openmm topology
     omm_bond_count = 0
 
-    # Create bonds preserving the bond ordering
-    for bond in mol.GetBonds():
+    # Creating bonds
+    for oe_bond in mol.GetBonds():
 
         omm_bond_count += 1
 
-        aromatic = None
-
-        # Set the bond aromaticity
-        if bond.IsAromatic():
-            aromatic = 'Aromatic'
+        # Set the bond type
+        if oe_bond.GetType() is not "":
+            if oe_bond.GetType() in ['Single', 'Double', 'Triple', 'Aromatic', 'Amide']:
+                omm_bond_type = oe_bond.GetType()
+            else:
+                omm_bond_type = None
+        else:
+            if oe_bond.IsAromatic():
+                oe_bond.SetType("Aromatic")
+                omm_bond_type = "Aromatic"
+            elif oe_bond.GetOrder() == 1:
+                oe_bond.SetType("Single")
+                omm_bond_type = "Single"
+            elif oe_bond.GetOrder() == 2:
+                oe_bond.SetType("Double")
+                omm_bond_type = "Double"
+            elif oe_bond.GetOrder() == 3:
+                oe_bond.SetType("Triple")
+                omm_bond_type = "Triple"
+            # Amide bond is missing ????
```

Comment:
@nividic : Does this mean OpenMM needs to add support for an "amide" bond type to support the standard OpenEye aromaticity model?

Reply:
If so, we need to get that into OpenMM ASAP before 7.2 is released!
Tagging @lnaden as well.

Reply:
Amide already exists in OpenMM: https://github.com/pandegroup/openmm/blob/master/wrappers/python/simtk/openmm/app/topology.py#L65-L68

Reply:
Hi there,

I’m sorry for my late reply. I made some improvements to the conversion functions between OpenEye <-> OpenMM. I mainly improved the bond type recognition. OpenMM supports “Single”, “Double”, “Triple”, “Aromatic” and “Amide” bond types (as spotted by @Lnaden). In the conversion function OpenEye->OpenMM I was able to check these bonds. However, it is not clear to me how OpenMM is perceiving bond types and bond orders. @jchodera  @Lnaden , If I try to load in a topology file by using OpenMM (I’m using OpenMM ver 7.1.1) PDBFile, AmberPrmtopFile the OpenMM bond types and orders are set to None. How can I perceive them with OpenMM? Therefore, the conversion function between OpenMM -> OpenEye needs improvements. I’m open to any suggestions and ideas.

Reply:
> I’m open to any suggestions and ideas.

So are we.

The OpenMM Topology does not try to infer the bond order from structure, only if the input topology file (AMBER promtop, GROMACS .top, Desmond DMS, etc.) has that information encoded in it. I also don't think OpenMM has taken a pass back through those readers to try and catch the ones which have the correct information and read it into the OpenMM `Topology` object. 

Having this conversion feature however is still a great thing though as we can actually start using the bond-order information more readily without worrying about information loss. 

@jchodera anything to add?

Reply:
These bond types were simply meant to be containers to permit round trips between OpenEye `OEMol` (and hopefully `RDKit` molecules) and OpenMM `Topology` objects. For now, you have to read small molecules using OpenEye to expect to populate these sensibly---none of the OpenMM readers currently populate these fields because none of the biopolymer file formats OpenMM can currently read (PDB, PDBx, mmCIF) carry this information (which is a major problem in our field).

Eventually, this feature will be used to enable Antechamber-based [residue template generators](http://docs.openmm.org/latest/userguide/application.html#adding-residue-template-generators) to generate GAFF parameters. We have a prototype of this in [`openmoltools`](https://github.com/choderalab/openmoltools/blob/master/openmoltools/forcefield_generators.py#L80-L179), but it currently uses a horrible hack to generate `OEMol`s from OpenMM `Residue` objects. Once we switch to your tools, we're hoping that it will allow us to correctly translate OpenMM `Residue` objects to `OEMol`s to have OpenEye-powered GAFF support for OpenMM